app/ts/background/windows/interceptorAccess.ts

```diff
@@ -114,212 +99,178 @@ export async function requestAccessFromUser(
 	website: Website,
 	request: InterceptedRequest | undefined,
 	requestAccessToAddress: AddressInfoEntry | undefined,
-	associatedAddresses: AddressInfoEntry[],
 	settings: Settings,
 ) {
-	const rejectReply = () => {
-		if (request) refuseAccess(websiteTabConnections, socket, request)
-	}
-
-	if (pendingInterceptorAccess !== undefined) return rejectReply()
-
+	let justAddToPending = false
+	if (pendingInterceptorAccess.size !== 0) justAddToPending = true
+	
 	// check if we need to ask address access or not. If address is put to never need to have address specific permision, we don't need to ask for it
 	const askForAddressAccess = requestAccessToAddress !== undefined && settings.userAddressBook.addressInfos.find((x) => x.address === requestAccessToAddress.address)?.askForAddressAccess !== false
 	const accessAddress = askForAddressAccess ? requestAccessToAddress : undefined
+	const future = new Future<InterceptorAccessReply>()
+	const requestId = request === undefined ? -Math.random() : request.requestId // if there's no particular request requesting this access, generate random ID for it
```

Comment:
Why not just a file-local constant that increments over time?  Does this need to be unique across runs or something?

Reply:
I'll realized it has to be consistent between sessions too. I'll keep it as random

Reply:
I also realized I can generate that Id deterministically :)